lib/StripeResource.js

```diff
@@ -258,7 +297,16 @@ StripeResource.prototype = {
       makeRequestWithData(null, utils.stringifyRequestData(data || {}));
     }
 
-    function makeRequest(apiVersion, headers) {
+    // add idempotency key if we're retrying the request
```

Comment:
I think this comment is confusing because it describes the behaviour when the function is called, not what the function does. It also implies we only add the header on the retry and not on the first attempt (which we should not do since it could cause 2 separate requests to be sent)

Reply:
Further comment nits, not really related: 

1) function docstrings should use block comment format (just `/**`, not like `@param` stuff). 
2) Comments should be in sentence case (first letter caps) with a period at the end (many comments in this PR... to be fair, I bet a lot of our comments don't follow this).